src/groupByKey.ts

```diff
@@ -2,6 +2,18 @@ type GroupsMap<T> = {
   [key: string]: T[];
 };
 
-export function groupByKey(items, key) {
-  // write code here;
+export function groupByKey<T>(items: T[], key: keyof T): GroupsMap<T> {
+  const result: GroupsMap<T> = {};
+
+  items.forEach((item: T) => {
+    const value = String(item[key]);
+
+    if (value in result) {
+      result[value].push(item);
+    } else {
+      result[value] = [item];
+    }
```

Comment:
```suggestion
    if (!result[value]) {
      result[value] = [];
    }
      result[value].push(item);
    
```

we can simplify it a bit and remove else block